### ISSUE TITLE
Lots of Errors, Warnings and such within LXMF.py

### ISSUE THREAD

**Issue by HarlekinSimplex** [markqvist/LXMF#3]
Hi Mark,
on my investigation regarding how I can get a notifcation from LXMF that a message has arrived at a destination, I took a deep look into the LXMF.py code.
I pulled the file into my PyCharm IDE and got quite a lot Warnings and code hints of all kind.
Some of them may be quite critical...

First: Line 233 
```
            # If no desired delivery method has been defined,
            # one will be chosen according to these rules:
            if self.desired_method == None:
                self.desired_method == LXMessage.DIRECT
            # TODO: Expand rules to something more intelligent
```
Guess here is the reason why leaving `desired_method` unset while calling `LXMessage()` does not work.
Instead of:
`                self.desired_method == LXMessage.DIRECT
`
I think it should state:
`                self.desired_method = LXMessage.DIRECT
`

Second: Line 314
```
    def determine_transport_encryption(self):
        if self.method == LXMessage.OPPORTUNISTIC:
            if self.__destination.type == RNS.Destination.SINGLE:
                self.transport_encrypted = True
                self.transport_encryption = LXMessage.ENCRYPTION_DESCRIPTION_EC
            elif destination_type == RNS.Destination.GROUP:
                self.transport_encrypted = True
                self.transport_encryption = LXMessage.ENCRYPTION_DESCRIPTION_AES
            else:
                self.transport_encrypted = False
                self.transport_encryption = LXMessage.ENCRYPTION_DESCRIPTION_UNENCRYPTED
```
Here the line:
`            elif destination_type == RNS.Destination.GROUP:
`
May probably should state ...
`            elif self.__destination_type == RNS.Destination.GROUP:
`
... since it is used later this way.
But in this case the object has no such member declared to judge on that assumption.
In general there are lots of members that are not declared explicitely, making reading the code even more cumbersome.

Third: Line 1269
```
        except Exception as e:
            RNS.log("Could not enable propagation node. The contained exception was: " + str(e), RNS.LOG_ERROR)
            raise e
            RNS.panic()
```
Here `RNS.panic()` is just dead code, that can not be reached,
And to use `raise e` is discouraged, because simply `raise` ususally puts the line into the log that contains the error and not the line of the raise statement as with `raise e`.

So probably the above snipped should be:
```
        except Exception as e:
            RNS.log("Could not enable propagation node. The contained exception was: " + str(e), RNS.LOG_ERROR)
            raise
```
Since I have no clue if the `RNS.panic()` was placed there with purpose, I cant really correct that one properly.

Forth: Line 1637
```
        for peer in culled_peers:
            RNS.log("Removing peer " + RNS.prettyhexrep(peer) + " due to excessive unreachability", RNS.LOG_WARNING)
            try:
                # TODO: Check this
                if peer_id in self.peers:
                    self.peers.pop(peer_id)
            except Exception as e:
                RNS.log(
                    "Error while removing peer " + RNS.prettyhexrep(peer_id) + ". The contained exception was: " + str(
                        e), RNS.LOG_ERROR)
```
Since PyCharm told me that `peer_id` within 
`                if peer_id in self.peers:
`
migth be accessd prior its initialized, the snippet might be changed into:
```
        for peer_id in culled_peers:
            RNS.log("Removing peer " + RNS.prettyhexrep(peer_id) + " due to excessive unreachability", RNS.LOG_WARNING)
            try:
                # TODO: Check this
                if peer_id in self.peers:
                    self.peers.pop(peer_id)
            except Exception as e:
                RNS.log(
                    "Error while removing peer " + RNS.prettyhexrep(peer_id) + ". The contained exception was: " + str(
                        e), RNS.LOG_ERROR)
```
Same as before, I'm not sure if thats what you intendet, especially because there is this ToDo for checking this there.
Thats where I stopped analysing all the hints and warnigs indicated by PyCharm.

And there is still lots more of other stuff, that may need some attention too:
- Unresoved attribute references
- Variable references before assignments
- Deprecated setDaemon() usage
- And dozens of other warnings and PEP 8: E711 code warnings

Is that file the most recent one?
If so why not putting some love into cleaning up that awsome piece of network code in a way, that it does not have that many warnigs and improvement hints any longer?

Cheers
Stephan

**Comment by markqvist** [comment]
Hi Stephan

> Instead of:
> ` self.desired_method == LXMessage.DIRECT`
> I think it should state:
> ` self.desired_method = LXMessage.DIRECT`

Thanks, that is indeed a typo. Fixed in 505176b6eef9e6e4915a87b8bc104da9811f1e4b.

> Here the line:
> ` elif destination_type == RNS.Destination.GROUP:`
> May probably should state ...
> ` elif self.__destination_type == RNS.Destination.GROUP:`
> ... since it is used later this way.
> But in this case the object has no such member declared to judge on that assumption.

Correct again. Fixed in 2d1828513faa1a7e7c0abf98cbcc2db14f2b4bf6. This is actually a remnant of skeleton code, that was not yet updated, since group destinations in LXMF is not yet implemented. But I'll get back to that. 

> Third: Line 1269
> Here `RNS.panic()` is just dead code, that can not be reached, And to use `raise e` is discouraged, because simply `raise` ususally puts the line into the log that contains the error and not the line of the raise statement as with `raise e`.

No, `RNS.panic()` gracefully terminates the stack and the currently attached program, which was the intention here. Whether that is a bit overkill could be argued. It made sense when the decision was made, but maybe we should just raise the exception instead.

Stylistically I think it makes more sense to explicitly `raise e`. Semantically it makes no difference, and is consistent with the style of everything else.

> Since PyCharm told me that `peer_id` within
> ` if peer_id in self.peers:`
> migth be accessd prior its initialized, the snippet might be changed

Yes, that is a typo, it should indeed be `peer_id`. Fixed in ed4effdab700094d3096dc36a61708e285a6b5f8.

Thanks a lot for spotting and taking the time to report these!

**Comment by HarlekinSimplex** [comment]
Yes. :-)
